### PR TITLE
[JARVIS-111] proper use of <f:password>

### PR DESCRIPTION
**Pull request by kohsuke**
This keeps the constructor and getter symmetric.

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/advisor/AdvisorGlobalConfiguration.java [hunk 1]
```diff
@@ -211,8 +211,8 @@ public void setEmail(@CheckForNull String email) {
     return email;
   }
 
-  public @Nonnull String getPassword() {
-    return Secret.toString(password);
+  public @Nonnull Secret getPassword() {
```
Comment (kwhetstone) [inline src/main/java/com/cloudbees/jenkins/plugins/advisor/AdvisorGlobalConfiguration.java]: I think this change ends up removing the password from the password field after the configuration is saved.  We moved to the String in order to have that password appear.

Comment (kohsuke) [inline src/main/java/com/cloudbees/jenkins/plugins/advisor/AdvisorGlobalConfiguration.java]: My expectation here, and I admit I haven't actually tried, is that <f:password> in the index.jelly will handle `Secret` correctly and "do the right thing", which is to put a scrambled value in the field that round trips.

**Comment by jtnord** [comment]
probably the validate function needs to be updated to take a Secret not a string.
https://github.com/jenkinsci/cloudbees-jenkins-advisor-plugin/blob/23190d027d5332357e8c979dcc9680aa933be2fc/src/main/java/com/cloudbees/jenkins/plugins/advisor/AdvisorGlobalConfiguration.java#L306

**Comment by kwhetstone** [comment]
I'll make an alternative PR with all the changes.  Thanks for all the help!

**Comment by Evildethow** [comment]
Closed in favor of https://github.com/jenkinsci/cloudbees-jenkins-advisor-plugin/pull/5

**Pull request closed without merging**